api/v1/storageclusterpeer_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type StorageClusterPeerState string
+
+const (
+	StorageClusterPeerFinalizer = "storageclusterpeer.ocs.openshift.io"
+)
+
+// StorageClusterPeerSpec defines the desired state of StorageClusterPeer
+type StorageClusterPeerSpec struct {
+
+	// APIServerEndpoint is the URI of the ocs api server
+	APIServerEndpoint string `json:"APIServerEndpoint"`
+
+	// OnboardingTicket holds an identity information required for provider to onboard.
+	OnboardingTicket string `json:"onboardingTicket"`
+}
+
+// StorageClusterPeerStatus defines the observed state of StorageClusterPeer
+type StorageClusterPeerStatus struct {
+
+	// Phase describes the Phase of StorageClusterPeer
+	// This is used by OLM UI to provide status information to the user
+	Phase StorageClusterPeerState `json:"phase,omitempty"`
```

Comment:
What are the possible values here?
Will this show if the connection to the peer cluster is lost?
Will this show when an existing peer needs to be re-boarded?

Reply:
For now, we can start with Initializing, Onboarding, Offboarding, Connected and Deleting
This will not show if the connection is lost.
I don't think we would need to re-board the existing peer

Reply:
I am noe sure you want phase here, more and more operators are moving away from this pattern. 
My suggestion is to have it empty for now and fill it in as the implementation goes forward and you identify meaningful information to add to the status

Reply:
I have removed the phase for now and will fill in the status with meaningful information as the implementation goes forward